server/jetty/src/main/java/io/deephaven/server/jetty/jsplugin/JsPlugins.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+ */
+package io.deephaven.server.jetty.jsplugin;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.jimfs.Configuration;
+import com.google.common.jimfs.Jimfs;
+import io.deephaven.plugin.type.JsPlugin;
+import io.deephaven.plugin.type.JsPluginRegistration;
+import org.eclipse.jetty.servlet.DefaultServlet;
+import org.eclipse.jetty.servlet.ServletHolder;
+
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.UncheckedIOException;
+import java.nio.file.FileSystem;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * A {@link JsPluginRegistration} for Jetty.
+ *
+ * @see #servletHolder(String)
+ */
+public final class JsPlugins implements JsPluginRegistration {
+
+    /**
+     * Creates a new js plugins instance with an in-memory filesystem.
+     *
+     * @return the js plugins
+     * @throws IOException if an I/O exception occurs
+     */
+    public static JsPlugins create() throws IOException {
+        final FileSystem fs = Jimfs.newFileSystem(Configuration.unix());
+        final JsPlugins jsPlugins = new JsPlugins(fs.getPath("/"));
+        jsPlugins.writeManifest(); // empty manifest
+        return jsPlugins;
+    }
+
+    private final Path path;
+    private final Map<String, JsPlugin> plugins;
+
+    private JsPlugins(Path path) {
+        this.path = Objects.requireNonNull(path);
+        this.plugins = new LinkedHashMap<>();
+    }
+
+    /**
+     * Creates a {@link DefaultServlet} servlet holder.
+     *
+     * @param name the name
+     * @return the servlet holder
+     */
+    public ServletHolder servletHolder(String name) {
+        final ServletHolder jsPlugins = new ServletHolder(name, DefaultServlet.class);
```

Comment:
I'm not sure what reasonable cache values are - I'll see if we are inheriting any cache values from jetty; and at a minimum, will add in etag support.

Reply:
The context itself has etags enabled, but I am not at all certain that is inherited by a sub-DefaultServlet. There are filters for caching, but they are explicit for files that include their hash in their name. Both of these are in JettyBackedGrpcServer.

Reply:
Idle thought on caching: at least for `main`, we could rename the file when copying, and report the renamed file via the `main` value in the json. The renamed string could include a hash in it, so that the url necessarily changes when the file changes, so the file can always be cached.

Same applies for the full plugin by naming the directory of the plugin with the hash of the contents, but the directory walking has to be deterministic.